crates/bevy_sprite/src/picking_backend.rs

```diff
@@ -0,0 +1,179 @@
+            .filter(|(_, camera, _, _)| camera.is_active)
+            .find(|(_, camera, _, _)| {
+                camera
+                    .target
+                    .normalize(Some(match primary_window.get_single() {
+                        Ok(w) => w,
+                        Err(_) => return false,
+                    }))
+                    .unwrap()
+                    == location.target
+            })
+        else {
+            continue;
+        };
+
+        let Some(cursor_ray_world) = camera.viewport_to_world(cam_transform, location.position)
+        else {
+            continue;
+        };
+        let cursor_ray_len = cam_ortho.far - cam_ortho.near;
+        let cursor_ray_end = cursor_ray_world.origin + cursor_ray_world.direction * cursor_ray_len;
+
+        let picks: Vec<(Entity, HitData)> = sorted_sprites
+            .iter()
+            .copied()
+            .filter(|(.., visibility)| visibility.get())
+            .filter_map(
+                |(entity, sprite, atlas, image, sprite_transform, pickable, ..)| {
+                    if blocked {
+                        return None;
+                    }
+
+                    // Hit box in sprite coordinate system
+                    let (extents, anchor) = if let Some((sprite, atlas)) = sprite.zip(atlas) {
+                        let extents = sprite.custom_size.or_else(|| {
+                            texture_atlas_layout
+                                .get(&atlas.layout)
+                                .map(|f| f.textures[atlas.index].size().as_vec2())
+                        })?;
+                        let anchor = sprite.anchor.as_vec();
+                        (extents, anchor)
+                    } else if let Some((sprite, image)) = sprite.zip(image) {
+                        let extents = sprite
+                            .custom_size
+                            .or_else(|| images.get(image).map(|f| f.size().as_vec2()))?;
+                        let anchor = sprite.anchor.as_vec();
+                        (extents, anchor)
+                    } else {
+                        return None;
+                    };
+
+                    let center = -anchor * extents;
+                    let rect = Rect::from_center_half_size(center, extents / 2.0);
+
+                    // Transform cursor line segment to sprite coordinate system
+                    let world_to_sprite = sprite_transform.affine().inverse();
+                    let cursor_start_sprite =
+                        world_to_sprite.transform_point3(cursor_ray_world.origin);
+                    let cursor_end_sprite = world_to_sprite.transform_point3(cursor_ray_end);
+
+                    fn inverse_lerp(a: f32, b: f32, x: f32) -> Option<f32> {
+                        if (b - a).abs() < f32::EPSILON {
+                            None
+                        } else {
+                            Some((x - a) / (b - a))
+                        }
+                    }
```

Comment:
I think this `filter_map` is busy enough without having helper functions inline, so I would personally move that out. Or maybe it already exists somewhere in the bevy codebase?

Reply:
I looked around but couldn't find it. I originally had it at the file level, but tbh it's so generic it just looked weird. I'd be in favor of stuffing it somewhere in `bevy_math` or something.

Reply:
I was thinking about where to put and thought "maybe this should actually be in glam". Well it is, here: https://docs.rs/glam/latest/glam/trait.FloatExt.html#tymethod.inverse_lerp

Reply:
Oh thanks! I checked glam but couldn't find it.